### PR TITLE
[TIMOB-1618] Android: Expose "minimumFontSize" on Android (Parity)

### PR DESCRIPTION
**Pull request by hansemannn**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-1618

Continuing from https://github.com/appcelerator/titanium_mobile/pull/5909, thank you @lgolu!

### CODE REVIEW

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUILabel.java
Comment (garymathews) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUILabel.java]: Could use `getFontProperties()` instead?

```Java
String[] fontProperties = TiUIHelper.getFontProperties(proxy.getProperties());
if (fontProperties.length > TiUIHelper.FONT_SIZE_POSITION && fontProperties[TiUIHelper.FONT_SIZE_POSITION] != null) {
    tv.setTextSize(TiUIHelper.getSizeUnits(fontProperties[TiUIHelper.FONT_SIZE_POSITION]), TiUIHelper.getSize(fontProperties[TiUIHelper.FONT_SIZE_POSITION]));
} else {
    tv.setTextSize(TiUIHelper.getSizeUnits(null), TiUIHelper.getSize(null));
}
```

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUILabel.java
Comment (garymathews) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUILabel.java]: Unnecessary brackets around `(tv.getText())`

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUILabel.java
Comment (garymathews) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUILabel.java]:
```Java
... && stringWidth > 0
```

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUILabel.java
Comment (garymathews) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUILabel.java]:
```Java
TiConvert.toFloat(minimumFontSize, 0)
```

**Comment by hansemannn** [comment]
Comments addressed!

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUILabel.java
Comment (garymathews) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUILabel.java]: This can be removed

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUILabel.java
Comment (garymathews) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUILabel.java]: This can be removed

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUILabel.java
Comment (garymathews) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUILabel.java]: This can be removed

**Comment by hansemannn** [comment]
@garymathews PR updated. Happy new year!

File: apidoc/Titanium/UI/Label.yml
Comment (fmerzadyan) [inline apidoc/Titanium/UI/Label.yml]: + place android part first in `since: {...}`.

File: apidoc/Titanium/UI/Label.yml
Comment (fmerzadyan) [inline apidoc/Titanium/UI/Label.yml]: keeping in format with existing docs:
+ make `ellipsize` into link to property link?
+ either edit into **or** append `default: <Titanium.UI.TEXT_ELLIPSIZE_TRUNCATE_END> on Android`...and so and so behavior on iOS, etc...

File: android/titanium/src/java/org/appcelerator/titanium/TiC.java
Comment (fmerzadyan) [inline android/titanium/src/java/org/appcelerator/titanium/TiC.java]: keeping in format with TiC:
+ properties are in alphabetical order so place after [PROPERTY_MIN_AGE](https://github.com/hansemannn/titanium_mobile/blob/d054b18df81e65f6b7ca165eb548476fb8c1e42a/android/titanium/src/java/org/appcelerator/titanium/TiC.java#L1890).
+ refactor to `PROPERTY_MIN_FONT_SIZE` like `PROPERTY_MIN_AGE`. not sure why `PROPERTY_MIN_AGE`, `PROPERTY_MIN_UPDATE_DISTANCE`, `PROPERTY_MIN_UPDATE_TIME` do not have `@module.api` tag.

File: apidoc/Titanium/UI/Label.yml
Comment (fmerzadyan) [inline apidoc/Titanium/UI/Label.yml]: + remove new line.

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUILabel.java
Comment (fmerzadyan) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUILabel.java]: + refactor variable to `view` or change `@param view` to `@param v`

**Review by frankieandone — CHANGES_REQUESTED** [review]
**Important note**: I suggest we add the same fix to [TextField](https://docs.appcelerator.com/platform/latest/#!/api/Titanium.UI.TextField)

File: apidoc/Titanium/UI/Label.yml
Comment (hansemannn) [inline apidoc/Titanium/UI/Label.yml]: Why?

File: android/titanium/src/java/org/appcelerator/titanium/TiC.java
Comment (hansemannn) [inline android/titanium/src/java/org/appcelerator/titanium/TiC.java]: It has to be `PROPERTY_MINIMUM_FONT_SIZE`. The constants are based on the camel case name that injects `_` chars instead of the camel-case style.

File: apidoc/Titanium/UI/Label.yml
Comment (hansemannn) [inline apidoc/Titanium/UI/Label.yml]: Yes, we can link it. But the default value suggestion should not be done, since the default is undefined, so behaving as not set. Otherwise it confuse the developers if they should set `TEXT_ELLIPSIZE_*` constants instead of font-sizes.

Comment (fmerzadyan) [inline apidoc/Titanium/UI/Label.yml]: sorry my bad, did review twice, first time added this comment. "lost" first review so inputted comments again then it looked like I had two reviews. Removed new and updated second review and this was left I guess from first time... :+1:

File: apidoc/Titanium/UI/Label.yml
Comment (fmerzadyan) [inline apidoc/Titanium/UI/Label.yml]: other examples always have android first. I think it's alphetical

Comment (hansemannn) [inline apidoc/Titanium/UI/Label.yml]: Nope, it's random. Usually the API added first comes first and the new one is added to it. But there're really no guidelines for that one.

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUILabel.java
Comment (fmerzadyan) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUILabel.java]: Also, it looks like the other comment before was removed so:
+ refactor parameter name to `view` or change `@param view` to `@param v`

**Comment by hansemannn** [comment]
Good idea to also implement it on the text field. @fmerzadyan do you want to take-over the PR? I just did some cleanup, the deep-dive should be done by an Android Pro :-)

**Comment by fmerzadyan** [comment]
@hansemannn haha ~ not that good but sure 👍

**Comment by fmerzadyan** [comment]
CR and FT passed

**Review by frankieandone — APPROVED** [review]

**Comment by lokeshchdhry** [comment]
FR Passed.
The `minimumFontSize` property works fine.
Checked it by code below:

````
var win = Ti.UI.createWindow({});

var label = Ti.UI.createLabel({
	text:"Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
	textColor:'white',
	minimumFontSize:8
});

win.add(label);
win.open();
````
1. Checked with different values for `minimumFontSize`.
2. Checked it in portrait & landscape mode.
3. Checked it for alloy app.
The text scales automatically to the minimum font size specified for the `minimumFontSize` & works as expected.

Appc Studio : 4.8.1.201612050850
SDK Version : local 6.1.0
Mac OS Version : 10.12.2
Xcode Version : Xcode 8.2.1 Build version 8C1002
Appc CLI AND Appc NPM : {"NPM":"4.2.9-1","CLI":"6.1.0"}
Ti CLI : 5.0.11
Alloy : 1.9.5
Node : v4.6.0
Nexus 6 running 6.0.1, Nexus 6P running 7.1.1 & android emulator 5.1.1

**Comment by lokeshchdhry** [comment]
@hansemannn , What about the `minimumFontSize` property for `textField` for android. Ae we gonna expose it as well ?

Will merge the PR once this is confirmed.

**Comment by hansemannn** [comment]
I think @m1ga talked about that a while ago, but for this specific 6.1.0 ticket, we will go with the labels first.

**Pull request merged**
